gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -352,5 +653,9 @@ void GazeboRosCamera::OnTrigger(const std_msgs::msg::Empty::SharedPtr)
   impl_->triggered++;
 }
 
-GZ_REGISTER_SENSOR_PLUGIN(GazeboRosCamera)
+extern "C" GZ_PLUGIN_VISIBLE gazebo::SensorPlugin * RegisterPlugin();
+gazebo::SensorPlugin * RegisterPlugin()
+{
+  return (gazebo::CameraPlugin *)(new GazeboRosCamera());
+}
```

Comment:
Why did you need to change this? Because of the multiple inheritance?

Reply:
Yes. Without the change, I was facing the diamond inheritance problem